### PR TITLE
network policies rego policy

### PR DESCRIPTION
**Pull request by danielmarzini**
Solves issue #9

### CODE REVIEW

File: gke-policies/policy/network_policies_test.rego [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package gke.policy.network_policies_engine
+
+test_dataplane_v1_without_netpol {
+  valid with input as {"name": "test-cluster",  "addons_config": { "network_policy_config": {"disabled": true }},"private_cluster_config": {"enable_private_nodes": true}}
```
Comment (rickygodoy) [inline gke-policies/policy/network_policies_test.rego:18]: Shouldn't this test the **not** valid?
In this example, the cluster has network policies disabled.

File: gke-policies/policy/network_policies.rego [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+dataplane_v1 {
+  dataplane_v1_netpol_config
+  dataplane_v1_netpol_enabled
+}
+
+dataplane_v2 {
+  not input.addons_config.network_policy_config.disabled
+  not input.network_config.datapath_provider == 2
```
Comment (rickygodoy) [inline gke-policies/policy/network_policies.rego:44]: I think it's worth a comment on what is magic number 2.

File: gke-policies/policy/network_policies_test.rego (hunk 1 above)
Comment (danielmarzini) [reply gke-policies/policy/network_policies_test.rego:18]: @rickygodoy you're totally right here, there's something strange. Let me do an entire re-check

File: gke-policies/policy/network_policies.rego [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+violation[msg] {
```
Comment (mikouaj) [inline gke-policies/policy/network_policies.rego:47]: I think we should have at least three violations (logical ORs) for the following:

* Network policy addon is not enabled AND 
* Network Policy config is not enabled AND
* dataplane v2 is not used

OR

* Network policy addon is enabled AND 
* Network Policy config is not enabled AND
* dataplane v2 is not used

OR

* Network policy addon is not enabled AND 
* Network Policy config is enabled AND
* dataplane v2 is not used

In all above cases we don't care about network policy settings if dataplane v2 is used.
I assumed also don't care about Network Policy provider setting

File: gke-policies/policy/network_policies_test.rego [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  not valid with input as {"name": "test-cluster", "addons_config": {"network_policy_config": {}},"private_cluster_config": {"enable_private_nodes": false}, "network_policy": {}}
+}
+
+test_dataplane_v1_with_netpol {
+  valid with input as {"name": "test-cluster", "addons_config": {"network_policy_config": {}},"private_cluster_config": {"enable_private_nodes": false}, "network_policy": {"provider": 1, "enabled": true}}
+}
+
+test_dataplane_v2_with_netpol {
+  valid with input as {"name": "test-cluster", "addons_config": { "network_policy_config": {"disabled": true }, "network_config": {"datapath_provider": 2}}}
+}
```
Comment (mikouaj) [inline gke-policies/policy/network_policies_test.rego:35]: I would also add test for the following:
* (not valid) Network Policy Addon is enabled but Network Policy Config is disabled
* (not valid) network policy addon is disabled but network policy config is enabled (shouldn't happen in real life, but who knows :))
* (valid) Network Policy Addon is enabled and Network Policy Config is enabled

File: gke-policies/policy/network_policies_test.rego [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+test_dataplane_v1_without_netpol_disabled {
+  not valid with input as {"name": "test-cluster", "addons_config": {"network_policy_config": {}},"private_cluster_config": {"enable_private_nodes": false}, "network_policy": {"provider": 1, "enabled": false}}
+}
+
+test_dataplane_v1_without_netpol_conf {
+  not valid with input as {"name": "test-cluster", "addons_config": {"network_policy_config": {}},"private_cluster_config": {"enable_private_nodes": false}, "network_policy": {}}
+}
+
+test_dataplane_v1_with_netpol {
+  valid with input as {"name": "test-cluster", "addons_config": {"network_policy_config": {}},"private_cluster_config": {"enable_private_nodes": false}, "network_policy": {"provider": 1, "enabled": true}}
```
Comment (mikouaj) [inline gke-policies/policy/network_policies_test.rego:30]: Should this be valid ?
Input has no Network Policy Addon enabled, isn't that prerequisite for network policy configuration?

**Review by mikouaj — CHANGES_REQUESTED** [review]

**Review by mikouaj — APPROVED** [review]
LGTM ! :-)

**Pull request merged**